components/credentials/src/main/kotlin/org/cloudfoundry/credhub/requests/CertificateGenerationRequestParameters.kt

```diff
@@ -84,9 +84,13 @@ class CertificateGenerationRequestParameters {
             isEmpty(country)
         ) {
             throw ParameterizedValidationException(ErrorMessages.MISSING_CERTIFICATE_PARAMETERS)
-        } else if (isEmpty(caName) && !selfSigned && !isCa) {
+        }
+
+        if (isEmpty(caName) && !selfSigned && !isCa) {
```

Comment:
This change makes the code harder to understand. Now we have to think about which conditional block we are entering. Before, it was clear from the structure that we can only enter 1.

I disagree with this. I don't know what "when" is in kotlin, I guess it would be better than this.

Reply:
I found this easier to understand than before, because there are also several independent conditionals below the if/else if/if structure that I changed. Now it's a series of conditionals.

Is it not clear that these are throwing exceptions?